### PR TITLE
Feat(when): optimize type

### PR DESCRIPTION
**Pull request by liaoliao666**

### CODE REVIEW

**Comment by antfu** [comment]
Can you revert the formatting changes? Thanks

**Comment by liaoliao666** [comment]
> Can you revert the formatting changes? Thanks

okay

File: packages/shared/when/index.ts [hunk 1]
```diff
@@ -1,76 +1,74 @@
-import { watch, WatchSource } from 'vue-demi'
-import { ConfigurableFlush, promiseTimeout } from '../utils'
-
-export interface WhenToMatchOptions extends ConfigurableFlush {
-  /**
-   * Milseconds timeout for promise to resolve/reject if the when condition does not meet.
```
Comment (antfu) [inline packages/shared/when/index.ts:6]: Any reason the jsdoc get removed?

File: packages/shared/utils/index.ts [hunk 2]
```diff
@@ -2,12 +2,14 @@ export * from './is'
...
+  throwOnTimeout = false,
+  reason = 'Timeout',
+): Promise<void> {
   return new Promise((resolve, reject) => {
-    if (throwOnTimeout)
-      setTimeout(() => reject(reason), ms)
-    else
-      setTimeout(resolve, ms)
+    if (throwOnTimeout) setTimeout(() => reject(reason), ms)
+    else setTimeout(resolve, ms)
```
Comment (antfu) [inline packages/shared/utils/index.ts:12]: Can you revert these formatting changes?

File: packages/shared/when/index.test.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ describe('when', () => {
 
       invoke(async() => {
         expect(r.value).toBe(0)
-        await when(r, { flush: 'sync' }).changedTimes(3)
+        await when(r).changedTimes(3)
```
Comment (antfu) [inline packages/shared/when/index.test.ts:30]: Wondering why this get removed?

**Comment by antfu** [comment]
Made a few changes, thank you.

**Pull request merged**